x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -243,13 +243,12 @@ public boolean isAvailableWithLicense(XPackLicenseState licenseState) {
         }
 
         // The model license does not matter, this is the highest licensed level
-        if (licenseState.isActive() && XPackLicenseState.isAllowedByOperationMode(
-            licenseState.getOperationMode(), License.OperationMode.PLATINUM, true)) {
+        if (licenseState.isAllowedByLicense(License.OperationMode.PLATINUM, true, true)) {
             return true;
         }
 
         // catch the rest, if the license is active and is at least the required model license
-        return licenseState.isActive() && License.OperationMode.compare(licenseState.getOperationMode(), licenseLevel) >= 0;
+        return licenseState.isAllowedByLicense(licenseLevel, true, false);
```

Comment:
Same as above.

Reply:
I doubt that we do *not* allow Trial license here. If the license is trial, it will be allowed by the code above this line. So allowTrial or not does not make a difference here. But it seems a bit funny to state it like this. Plus it could be simplified to `isAllowedByLicense(licenseLevel)` if allowTrial is true.

Reply:
It feels like this whole method is trying to hard to duplicate logic that should be handled XPackLicenseState.
I think we can replace the whole method with: 
```
return licenseState.isAllowedByLicense(this.licenseLevel)
```

But perhaps we should make that a separate PR so that the ML team can review it.

Reply:
> I'm OK with that, but I worry that it will end up with silly comments because people feel weird about just `return true;`
> 

Fair point. I think I'll leave it as is for now. The actual improvement should be removing these checks from the caller site when we can be sure they are available in any case. We could take care of this change separately.